policy/decompressor.go

```diff
@@ -0,0 +1,133 @@
+package policy
+
+import (
+	"sync"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3metrics/generated/proto/schema"
+	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/log"
+)
+
+// Decompressor maintains a mapping of integers to policies. It is used to return a Policy from
+// its corresponding integer encoding
+type Decompressor interface {
+	Get(id int64) (Policy, bool)
+}
+
+type decompressorMap map[int64]Policy
+
+type noopDecompression struct{}
+
+// NewNoopDecompressor returns a new Decompressor mapping which never matches an id
+func NewNoopDecompressor() Decompressor {
+	return noopDecompression{}
+}
+
+func (m noopDecompression) Get(id int64) (Policy, bool) {
+	return EmptyPolicy, false
+}
+
+type staticDecompressor struct {
+	policies decompressorMap
+}
+
+// NewStaticDecompressor returns a new static Decompressor
+func NewStaticDecompressor(policies map[int64]Policy) Decompressor {
+	return staticDecompressor{policies: policies}
+}
+
+func (m staticDecompressor) Get(id int64) (Policy, bool) {
+	p, ok := m.policies[id]
```

Comment:
Primarily for tests, but I think it also could be useful in the case where you only want to support a fixed number of policies or are willing to restart the processes when the policies change. Dynamic policies require an external key-value store so if someone wants to opt out of that but still support policies they can with the static compressor.